### PR TITLE
#344 Handle Switch disabled state visually

### LINKED ISSUE #344: [Switch] Lock icon and cursor not-allowed for disabled state
## Description
`Switch` component should have a `lock` icon and
```css
cursor: not-allowed;
```
in case of a `disabled` state.

## Design
[Figma - Locked Switch](https://www.figma.com/file/2pFu80PXO5A2tfyrAGnx91/Product-Components?node-id=6904%3A32302)

image

### PR DESCRIPTION
**Pull request by sgraczyk**
Resolves #344 

## Description
- Lock icon added for the disabled state,
- Applied proper color tokens for `hover`, `enabled`, and `disabled` cases,
- Changed naming convention as `enabled/disabled` was easy to be confused with `checked (on)/unchecked (off)`,
- Added multiple stories representing the changes,
- I've used `--color-action-default-rgb` for `box-shadow`

## Storybook
- [Disabled On](https://613a8e945a5665003a05113b-muqgbybjhw.chromatic.com/?path=/story/components-switch--disabled-on)
- [Disabled Off](https://613a8e945a5665003a05113b-muqgbybjhw.chromatic.com/?path=/story/components-switch--disabled-off)

### CODE REVIEW

**Review by mreszke — APPROVED** [review]

File: packages/react-components/src/components/Switch/Switch.stories.tsx [hunk 1]
```diff
@@ -1,23 +1,35 @@
...
+EnabledOn.args = { ...basicWithoutStateHandler, on: true };
+
+export const EnabledOff = Template.bind({});
+EnabledOff.args = { ...basicWithoutStateHandler, on: false };
+
+export const DisabledOn = Template.bind({});
+DisabledOn.args = { ...basicWithoutStateHandler, on: true, disabled: true };
+
+export const DisabledOff = Template.bind({});
+DisabledOff.args = { ...basicWithoutStateHandler, on: false, disabled: true };
```
Comment (marcinsawicki) [inline packages/react-components/src/components/Switch/Switch.stories.tsx:35]: I think we can minimize the number of different stories here. We can divide it into "kinds" and "states", just like we have in `Button` component stories.

**Review by marcinsawicki — APPROVED** [review]

**Comment by marcinsawicki** [comment]
I little comment from me.

File: packages/react-components/src/components/Switch/Switch.stories.tsx (hunk 1 above)
Comment (sgraczyk) [reply packages/react-components/src/components/Switch/Switch.stories.tsx]: Addressed in https://github.com/livechat/design-system/pull/353/files/b1ba318006993158ba0bbe83499b5172ff99d8fd..7b2f4ecdd7189efd1a793d7d4fc2bbda4339a3a0

Comment (sgraczyk) [reply packages/react-components/src/components/Switch/Switch.stories.tsx]: @marcinsawicki Please check if this is what you've expected https://github.com/livechat/design-system/pull/353/commits/e443d1288ea431e9ba211fd8e717612c0ad8c826

Comment (marcinsawicki) [reply packages/react-components/src/components/Switch/Switch.stories.tsx]: It's perfect 👍

**Pull request merged**